### DISCUSSION TITLE
Access Nuxt 3 AsyncData pending variable globally without using store

### DISCUSSION THREAD

**Discussion by toddpadwick** [nuxt/nuxt#25446] [Questions]
I have a site which has an animation applied to the logo in the header, when a page is loading. This is especially useful when using AsnycData instead of lazyAsyncData whereby the page is not mounted until the asyncData function is complete. (which means no page level loading icon based on the pending variable will ever be seen).

So I need a way to globalise the pending variable of all page asyncData functions so I can trigger the logo animation loop based on whether they are true or not.

// [slug].vue
```
const {
    pending, // Explose this globally so I can access it in the header.vue component.
    data: entry,
    error
} = await useAsyncData("article_"+route.path, () => sanity.fetch(query));
```

I am aware I could add a loading variable to the Pinia store, and watch for pending on any given asyncData function. But this is not only clunky as it means adding a watcher to loads of pages manually, but it's also risky as race conditions could cause them to overlap.

Is there an efficient way of doing this? For example, using Apollo in Nuxt 2, doing so was rarely straightforward.

**Comment by TheAlexLichter** [comment]
So you want an info about *all* possible `pending` states?

**Comment by toddpadwick** [comment]
Hey @manniL  yes that's correct - so anytime _any_ asyncData function is pending, I would like the logo in the header banner component to animate.
